src/TicketReader.java

```diff
@@ -0,0 +1,78 @@
+import java.nio.file.*;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TicketReader {
+
+    public List<BusTicket> readTicketsFromFile(String fileName) {
+        List<BusTicket> tickets = new ArrayList<>();
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+
+        try {
+            List<String> lines = Files.readAllLines(Paths.get(fileName));
+
+            for (String line : lines) {
+                String[] parts = line.replace("{", "").replace("}", "").replace("\"", "").split(",");
+
+                if (parts.length >= 4) {
+                    String ticketClass = null;
+                    String ticketType = null;
+                    String startDateStr = null;
+                    Double price = null;
+
+                    for (String part : parts) {
+                        String[] keyValue = part.split(":");
+
+                        if (keyValue.length == 2) {
+                            String key = keyValue[0].trim();
+                            String value = keyValue[1].trim();
+
+                            switch (key) {
+                                case "ticketClass":
+                                    ticketClass = !value.equals("null") ? value : null;
+                                    break;
+                                case "ticketType":
+                                    ticketType = !value.equals("null") ? value : null;
+                                    break;
+                                case "startDate":
+                                    startDateStr = !value.equals("null") ? value : null;
+                                    break;
+                                case "price":
+                                    try {
+                                        price = !value.equals("null") && !value.isEmpty() ? Double.parseDouble(value) : null;
+                                    } catch (NumberFormatException e) {
+                                        System.out.println("Invalid price format: " + value);
+                                    }
+                                    break;
+                            }
```

Comment:
why java version you use?

Reply:
i am using Java 21, should i used old versions?

Reply:
@gujaKhela  Its ok)) but you use old style)) https://www.baeldung.com/java-switch#1-the-new-switch-expression
try use new one please